### PR TITLE
Set TokenValidationResult.TokenType with the token type returned by Validators.ValidateTokenType

### PR DESCRIPTION
**Pull request by kevinchalet**
If you approve the approach, I'll try to add a unit test.

/cc @brentschmaltz

### CODE REVIEW

File: src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs [hunk 1]
```diff
@@ -1146,7 +1146,8 @@ public virtual TokenValidationResult ValidateToken(string token, TokenValidation
                     {
                         SecurityToken = jwtToken,
                         ClaimsIdentity = innerTokenValidationResult.ClaimsIdentity,
-                        IsValid = true
+                        IsValid = true,
+                        TokenType = innerTokenValidationResult.TokenType
```
Comment (kevinchalet) [inline src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs:1150]: Not 100% sure about this one. Although I would expect the two to match in most cases, Wilson seems to make a difference between the `typ` of the JWE container and the `typ` of the inner JWT token, so maybe it should be another property, like `InnerTokenType`?

Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs:1150]: I think it should be as you mentioned the inner, as the JWE is just a wrapper without any info.

**Comment by brentschmaltz** [comment]
@kevinchalet I agree with your approach, feel free to add the tests.
fyi, we will probably put another release out next week.

**Comment by kevinchalet** [comment]
I updated the existing type validation tests to validate `TokenValidationResult.TokenType`.

I also opened https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1397 to track a possible improvement for token creation, that would make the story nicer and consistent with this PR.

**Pull request merged**